qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -127,43 +81,27 @@ def set_sweep(self, start, stop, npts):
         self.setpoints = (f,)
         self.shape = (npts,)
 
-    def get(self):
-        if not self._instrument._parent.rf_power():
-            log.warning("RF output is off when getting mag")
-        # it is possible that the instrument and qcodes disagree about
-        # which parameter is measured on this channel
-        instrument_parameter = self._instrument.vna_parameter()
-        if instrument_parameter != self._instrument._vna_parameter:
-            raise RuntimeError("Invalid parameter. Tried to measure "
-                               "{} got {}".format(self._instrument._vna_parameter,
-                                                  instrument_parameter))
-        self._instrument.write('SENS{}:AVER:STAT ON'.format(self._channel))
-        self._instrument.write('SENS{}:AVER:CLE'.format(self._channel))
-        self._instrument._parent.cont_meas_off()
-
-        # instrument averages over its last 'avg' number of sweeps
-        # need to ensure averaged result is returned
-        for avgcount in range(self._instrument.avg()):
-            self._instrument.write('INIT{}:IMM; *WAI'.format(self._channel))
-        data_str = self._instrument.ask(
-            'CALC{}:DATA? FDAT'.format(self._channel))
-        data = np.array(data_str.rstrip().split(',')).astype('float64')
-        if self._instrument.format() in ['Polar', 'Complex',
-                                         'Smith', 'Inverse Smith']:
-            log.warning("QCoDeS Dataset does not currently support Complex "
-                        "values. Will discard the imaginary part.")
-            data = data[0::2] + 1j * data[1::2]
-        self._instrument._parent.cont_meas_on()
-        return data
-
+    def get_raw(self):
+       data = self._instrument._get_sweep_data()
+       if self._instrument.format() in ['Polar', 'Complex',
+                            'Smith', 'Inverse Smith']:
+           log.warning("QCoDeS Dataset does not currently support Complex "
+                       "values. Will discard the imaginary part.")
+       return data
 
 class ZNBChannel(InstrumentChannel):
 
-    def __init__(self, parent, name, channel):
+    def __init__(self, parent, name, channel, vna_parameter: str=None):
+        """
+        Args:
+            vna_parameter: if left empty name is used
```

Comment:
I would write a bit more here.
```
Args:
    parent: Instrument that this channel is bound to.
    name: Name to use for this parameter
    vna_parameter: Name of parameter on the vna that this should measure such as S11. If left
               empty this will fall back to `name` 
```

Reply:
@ThorvaldLarsen could you write this? I think you can find more concise words than me ;-)

Reply:
Might just want to change a word or two for clarity. But the part that Jens wrote looks pretty good to me.
Args:
    parent: Instrument that this channel is bound to.
    name: Name to use for this channel.
    vna_parameter: Name of parameter on the vna that this should measure such as S12. If left
               empty this will fall back to `name`.

Reply:
done. Ready to merge?